robonomics_liability/src/robonomics_liability/executor.py

```diff
@@ -45,68 +45,48 @@ def incoming_liability(msg):
         rospy.Subscriber('incoming', Liability, incoming_liability)
 
         def finish_liability(msg):
-            self.liability_success = msg.success
-            self.liability_finish = True
+            liability_thread = self.liability_execution_threads.pop(msg.address)
+
+            liability_msg = liability_thread.getLiabilityMsg()
+            result = liability_thread.finish(msg.success)
+
+            self.complete.publish(liability_msg)
+            self.result_topic.publish(result)
+            rospy.loginfo('Liability %s finished with %s', liability_msg.address, result.result)
             return FinishLiabilityResponse()
         rospy.Service('finish', FinishLiability, finish_liability)
 
+        def start_liability(msg):
+            try:
+                liability_thread = self.liability_execution_threads[msg.address]
+            except Exception as e:
```

Comment:
Is possible to use `KeyError`: the KeyError is raised when accessing an invalid key within a dict.